vscode/quint-vscode/server/src/server.ts

```diff
@@ -274,54 +277,53 @@ export class QuintLanguageServer {
     })
 
     connection.onCompletion((params: CompletionParams): CompletionItem[] => {
-      // Only complete on "."
-      // TODO: offer completion at arbitrary positions in the text
-      if (params.context?.triggerKind !== CompletionTriggerKind.TriggerCharacter) {
-        return []
-      }
+      logger.debug(`Completion requested at position ${JSON.stringify(params.position)} in ${params.textDocument.uri}`)
 
-      const parsedData = this.parsedDataByDocument.get(params.textDocument.uri)
-      if (!parsedData) {
+      if (params.context?.triggerKind !== CompletionTriggerKind.Invoked) {
+        logger.debug(`Completion triggered by non-invoked action, ignoring.`)
         return []
       }
 
+      const parsedData = this.parsedDataByDocument.get(params.textDocument.uri)
       const document = this.documents.get(params.textDocument.uri)
-      if (!document) {
+
+      if (!parsedData || !document) {
+        logger.debug(`No parsed data or document found for completion.`)
         return []
       }
 
-      // Parse the identifier that triggered completion
-      // TODO: offer completion for chained calls
-      //       (e.g., `tup._2.`, `rec.field.` or `set.powerset().`)
       const triggeringLine = document.getText({
         start: { ...params.position, character: 0 },
         end: params.position,
       })
 
       const match = triggeringLine?.trim().match(/([a-zA-Z][a-zA-Z0-9_]*|[_][a-zA-Z0-9_]+)\.$/)
       if (!match) {
+        logger.debug(`No matching identifier found in line: ${triggeringLine}`)
         return []
       }
-      const triggeringIdentifier = match[1]
 
+      const triggeringIdentifier = match[1]
       const sourceFile = URI.parse(params.textDocument.uri).path
 
-      // Run analysis synchronously to get the required information for completion
       this.triggerAnalysis(document)
```

Comment:
Adding the comment back, I think it's useful
```suggestion
      // Run analysis synchronously to get the required information for completion
      this.triggerAnalysis(document)
```